vpr/src/place/place.cpp

```diff
@@ -1409,14 +1427,26 @@ static e_move_result try_swap(float t,
         rlim = std::numeric_limits<float>::infinity();
     }
 
+    if (manual_move) {
+        //pops up the manual move window for the user to input set their move
+        manual_move_generator_window("");
+        update_screen(ScreenUpdatePriority::MAJOR, " ", PLACEMENT, nullptr);
+        manual_move_info = get_manual_move_info();
+        //sends info to the move generator class
+        mmg_get_manual_move_info(*manual_move_info);
+    }
+
     //Generate a new move (perturbation) used to explore the space of possible placements
-    e_create_move create_move_outcome = move_generator.propose_move(blocks_affected, rlim);
+    e_create_move create_move_outcome;
+    if (!manual_move)
+        create_move_outcome = move_generator.propose_move(blocks_affected, rlim);
+    else
+        create_move_outcome = manual_move_generator.propose_move(blocks_affected, rlim);
 
     LOG_MOVE_STATS_PROPOSED(t, blocks_affected);
-
     e_move_result move_outcome = ABORTED;
 
-    if (create_move_outcome == e_create_move::ABORT) {
+    if (create_move_outcome == e_create_move::ABORT || !manual_move_info->valid_input) {
```

Comment:
Could have manual_moves return the ABORT type if the input is illegal, to simplify this if statement.  I think that would be better.